scripts/release.sh

```diff
@@ -0,0 +1,198 @@
+
+for arg in "$@"; do
+    case $arg in
+        --dry-run)
+            DRY_RUN=true
+            shift
+            ;;
+        major|minor|patch)
+            VERSION_TYPE="$arg"
+            shift
+            ;;
+        *)
+            echo "Error: Unknown argument '$arg'"
+            echo "Usage: $0 [major|minor|patch] [--dry-run]"
+            exit 1
+            ;;
+    esac
+done
+
+# Default to patch if no version type specified
+VERSION_TYPE="${VERSION_TYPE:-patch}"
+
+if [[ ! "$VERSION_TYPE" =~ ^(major|minor|patch)$ ]]; then
+    echo "Error: Version type must be 'major', 'minor', or 'patch'"
+    echo "Usage: $0 [major|minor|patch] [--dry-run]"
+    exit 1
+fi
+
+if [ "$DRY_RUN" = true ]; then
+    echo "[dry-run] - No changes will be made"
+    echo ""
+else
+    echo "Starting automated release process..."
+fi
+
+# Check for uncommitted changes (skip in dry-run)
+if [ "$DRY_RUN" = false ] && ! git diff-index --quiet HEAD --; then
+    echo "Error: You have uncommitted changes. Please commit or stash them first."
+    exit 1
+fi
+
+# Extract current version from CMakeLists.txt
+echo "Step 1: Reading current version..."
+MAJOR=$(grep "set(GGML_VERSION_MAJOR" CMakeLists.txt | sed 's/.*MAJOR \([0-9]*\).*/\1/')
+MINOR=$(grep "set(GGML_VERSION_MINOR" CMakeLists.txt | sed 's/.*MINOR \([0-9]*\).*/\1/')
+PATCH=$(grep "set(GGML_VERSION_PATCH" CMakeLists.txt | sed 's/.*PATCH \([0-9]*\).*/\1/')
+
+echo "Current version: $MAJOR.$MINOR.$PATCH-dev"
+
+# Calculate new version
+case $VERSION_TYPE in
+    major)
+        NEW_MAJOR=$((MAJOR + 1))
+        NEW_MINOR=0
+        NEW_PATCH=0
+        ;;
+    minor)
+        NEW_MAJOR=$MAJOR
+        NEW_MINOR=$((MINOR + 1))
+        NEW_PATCH=0
+        ;;
+    patch)
+        NEW_MAJOR=$MAJOR
+        NEW_MINOR=$MINOR
+        NEW_PATCH=$((PATCH + 1))
+        ;;
+esac
+
+NEW_VERSION="$NEW_MAJOR.$NEW_MINOR.$NEW_PATCH"
+echo "New release version: $NEW_VERSION"
+echo ""
+
+# Update CMakeLists.txt for release
+echo "Step 2: Updating version in CMakeLists.txt..."
+if [ "$DRY_RUN" = true ]; then
+    echo "  [dry-run] Would update GGML_VERSION_MAJOR to $NEW_MAJOR"
+    echo "  [dry-run] Would update GGML_VERSION_MINOR to $NEW_MINOR"
+    echo "  [dry-run] Would update GGML_VERSION_PATCH to $NEW_PATCH"
+    echo "  [dry-run] Would remove -dev suffix"
+else
+    sed -i "s/set(GGML_VERSION_MAJOR [0-9]*)/set(GGML_VERSION_MAJOR $NEW_MAJOR)/" CMakeLists.txt
+    sed -i "s/set(GGML_VERSION_MINOR [0-9]*)/set(GGML_VERSION_MINOR $NEW_MINOR)/" CMakeLists.txt
+    sed -i "s/set(GGML_VERSION_PATCH [0-9]*)/set(GGML_VERSION_PATCH $NEW_PATCH)/" CMakeLists.txt
+    sed -i 's/set(GGML_VERSION_DEV "-dev")/set(GGML_VERSION_DEV "")/' CMakeLists.txt
+fi
+echo ""
+
+# Commit version bump
+echo "Step 3: Committing version bump..."
+if [ "$DRY_RUN" = true ]; then
+    echo "  [dry-run] Would commit: 'ggml : bump version to $NEW_VERSION'"
+else
+    git add CMakeLists.txt
+    git commit -m "ggml : bump version to $NEW_VERSION"
```

Comment:
Since we have push to master blocked all changes go through PRs. So here it would be better to create a new branch, f.ex. `ggml-rc-vX.Y.Z`.

Also the script should check that the release is being made from latest `master` and print a warning if it is not.

Reply:
I've updated the script now so that it creates a branch, and also added the checks to make sure the current branch is master and has the latest.